ktor-client/ktor-client-cio/jvm/src/io/ktor/client/engine/cio/Endpoint.kt

```diff
@@ -146,8 +151,23 @@ internal class Endpoint(
 
                 if (address.isUnresolved) throw UnresolvedAddressException()
 
-                val connection = withTimeoutOrNull(connectTimeout) { connectionFactory.connect(address) }
-                    ?: return@repeat
+                val connection = when (connectTimeout) {
+                    HttpTimeout.INFINITE_TIMEOUT_MS -> connectionFactory.connect(address) {
```

Comment:
Consider extracting connect into single method(like in CIO Socket)

Reply:
Done. Moved into lambda because it's not possible to move it into pure function.